hibikase/PlaybackWidget.cpp

```diff
@@ -51,22 +50,46 @@ void PlaybackWidget::LoadAudio(std::unique_ptr<QIODevice> io_device)
 {
     if (m_worker)
     {
-        QMetaObject::invokeMethod(m_worker, "Stop");  // Ensure state gets set to stopped
+        disconnect(m_worker, &AudioOutputWorker::StateChanged, this, &PlaybackWidget::OnStateChanged);
+        disconnect(m_worker, &AudioOutputWorker::TimeUpdated, this, &PlaybackWidget::UpdateTime);
+        QMetaObject::invokeMethod(m_worker, "Stop");
```

Comment:
The reason I added a call to Stop here was to ensure that OnStateChanged gets called with StoppedState. Is there a point in having this line after the disconnect calls?

Reply:
I don't want the stop event to overwrite the button text :p

Reply:
Yes, but can't you remove this line entirely?

Reply:
In theory, but will the audio worker be destroyed quickly?

Reply:
It will be destroyed as quickly as this call to Stop would happen.